version1/06 - Access control.md

```diff
@@ -0,0 +1,217 @@
+
+Supported explicit permissions for the `all` code list type are all explicit permissions listed above.
+
+Supported explicit permissions for the `administrativenhet`, `mappetype`, `dokumenttype`, and `skjerming` code list types (and their list values) are only `ReadRelated`.
+
+### Explicit permissions use cases
+
+#### Users with global read access to the system
+
+To grant users access to read everything in the archive, create or update their access group with the following explicit/global permissions:
+
+- `ReadThis`
+- `Read`
+- `ReadRelated`
+
+#### Global administrators of the system
+
+To grant users access to read and edit everything in the archive, create or update their access group with the following explicit/global permissions:
+
+- `ReadThis`
+- `Read`
+- `Create`
+- `Update`
+- `UpdateSystemManaged`
+
+You could optionally grant `Delete`, but Documaster recommends that only a small number of people (if any) have rights to delete data in the archive.
+
+You can add `ReadRelated` in case there are any access controlling Klasser or Code lists.
+
+#### Global administrators of the system with rights to grant or revoke permissions to resources
+
+To grant users access to read and edit everything in the archive, including permissions on resources, create or update their access group with the following explicit/global permissions:
+
+- `ReadThis`
+- `Read`
+- `Create`
+- `Update`
+- `UpdateSystemManaged`
+- `Grant`
+
+You could optionally grant `Delete`, but Documaster recommends that only a small number of people (if any) have rights to delete data in the archive.
+
+You can add `ReadRelated` in case there are any access controlling Klasser or Code lists.
+
+#### Users with access only to a particular Fonds in the archive
+
+To grant users access to read and edit everything only in a single Fonds in the archive, create or update their access group with the following explicit/global permissions:
+
+- `ReadThis`
+
+Additionally, update the Fonds you would like to grant them access to like so:
+
+```json
+POST rms/api/public/noark5/v1/permission/entity
+
+{
+    "groupId": integer,
+    "objectType": "Arkiv",
+    "objectId": string,
+    "explicitPermissions": ["ReadThis", "Read", "Create", "Update", "UpdateSystemManaged"]
+}
+```
+
+You could optionally grant `Delete` on the Fonds, but Documaster recommends that only a small number of people (if any) have rights to delete data in the archive.
+
+Additionally, you will need to repeat the same request for all KlassifikasjonsSystemer to which the user needs to have access:
+
+```json
+POST rms/api/public/noark5/v1/permission/entity
+
+{
+    "groupId": integer,
+    "objectType": "Klassifikasjonssystem",
+    "objectId": string,
+    "explicitPermissions": ["ReadThis", "Read"]
+}
+```
+
+You could optionally grant `Create`, `Update`, or `Delete` for the Klassifikasjonssystemer if you would like users to be able to create, edit, or delete existing Klasser.
```

Comment:
Typo: Klassifikasjonssystemer -> Klassifikasjonssystem?

Reply:
It's actually on purpose.

I wanted to use plural, but Klassifikasjonssystems sounded weird when the word is written in Norwegian.

Klassifikasjonssystemer means Classification systems (plural) in Norwegian.